pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -1711,7 +1711,11 @@ type APIServerConfig struct {
 // +k8s:openapi-gen=true
 type AlertmanagerEndpoints struct {
 	// Namespace of the Endpoints object.
-	Namespace string `json:"namespace"`
+	// If not set, the object will be discovered in the namespace of the Prometheus object.
+	// +kubebuilder:validation:MinLength:=1
+	// +required
```

Comment:
```suggestion
	// +optional
```

Reply:
But the issue is with endpoint namespace

Reply:
The namespace field was "optional" before in the sense that you could set it to an empty value. Making it required would mean a breaking change.